quickwit-search/src/root.rs

```diff
@@ -65,13 +65,17 @@ pub async fn root_search(
     .map_err(|err| {
         SearchError::InternalError(format!("Failed to build doc mapper. Cause: {}", err))
     })?;
+
+    // try to build query against current schema
+    let _query = doc_mapper.query(doc_mapper.schema(), search_request)?;
+
     let doc_mapper_str = serde_json::to_string(&doc_mapper).map_err(|err| {
         SearchError::InternalError(format!("Failed to serialize doc mapper: Cause {}", err))
     })?;
     let split_metadata_list = list_relevant_splits(search_request, metastore).await?;
     let split_metadata_map: HashMap<String, SplitMetadata> = split_metadata_list
         .into_iter()
-        .map(|metadata| (metadata.split_id().to_string(), metadata))
+        .map(|mut metadata| (std::mem::take(&mut metadata.split_id), metadata))
```

Comment:
Note: no need to copy here

Reply:
Let's not do this. This probably has no impact at all on performance and it is harmful:
- the code is less simple and less readable
- you are modifying an object that was immutable.

Consumer of the metadata will  assume that it is properly populated and find an empty string.
Note that this value can be very handy for debugging for instance.

Reply:
Yeah you are right. Let me revert this :heart: